packages/vscode-extension/src/debug/microsoftTunnelingTask.ts

```diff
@@ -73,14 +74,14 @@ export class MicrosoftTunnelingTaskTerminal implements vscode.Pseudoterminal {
   private async openAsync(): Promise<Result<void, FxError>> {
     // TODO: add telemetry
     // TODO: prevent re-entry (cases when user manually trigger a task)
-    this.manager = new MicrosoftTunnelingManager(async () => {
-      // TODO: switch to new login and pass in Microsoft tunneling scopes
-      const token = await appStudioLogin.getAccessToken();
-      if (!token) {
-        throw new Error("No login");
-      }
-      return token;
-    }, this.logger);
+    this.manager = new MicrosoftTunnelingManager(
+      () =>
+        m365Login.getAccessToken({
+          showDialog: true,
```

Comment:
Interactive login on starting task may not be a good UX... any plan to move login to pre-checker?
Better to silently login here.

Reply:
I think prereqs check already ensures M365 account is signed in. Is this correct?

If so, this will succeed in most cases and won't pop-up user interaction (whether login or consent).
Because we are first party app and preauthorized for tunneling service, [acquiring token for this scope will not require user interaction](https://github.com/AzureAD/microsoft-authentication-library-for-js/blob/dev/lib/msal-browser/docs/resources-and-scopes.md#dynamic-scopes-and-incremental-consent). 

For the rare cases (e.g. token expired between prereqs check and tunneling), I agree changing to `showDialog: false` is better. But no need to add more checks to prereqs-check.